### DISCUSSION TITLE
change tutor7pp to score by history

### DISCUSSION THREAD

**Discussion by Jaap1957** [nrc-cnrc/EGSnrc#817] [Questions] [answered]
Hi,
I would like to do something like tutor7pp does, get energy fractions per region, but i would like to get them for each history.
Does someone have an example which gives output on a history basis?
Thanks !

**Comment by Jaap1957** [comment]
I found a solution.

I added some lines after line 711 of tutor7pp.cpp .
```
int Tutor7_Application::startNewShower() {
    Etot += p.E*p.wt;
    int res = EGS_Application::startNewShower();
    if (res) {
        return res;
    }
    if (current_case != last_case) {
        
        // Report edep for each region, per history
        cout <<"\nHistory; "<< last_case << "; EDEP; ";
        for (int j=0; j<nph; j++) {
            int ireg = ph_regions[j];
            EGS_Float edep = score->currentScore(ireg+1);
            cout <<  edep << "; ";
        }
        cout << "\n";
        if (nph > 0) {
```
This produces lines with the dose per voxel per history in the output.

Is there a way to get the edep output to a binary stream?

Thanks !

**Reply by rtownson** [reply]
This is a c++ question mostly, you would initialize a binary stream elsewhere in the code and then write to it instead of cout. Here's a little tutorial:
https://www.tutorialspoint.com/reading-and-writing-binary-file-in-c-cplusplus

**Comment by Jaap1957** [comment, marked as answer]
Indeed, it works like this:

1 add at the class:
`class APP_EXPORT Tutor7_Application : public EGS_AdvancedApplication {`
`ofstream     RH_stream;`         

2 add RH_stream(0) in the definition of the application:

` Tutor7_Application(int argc, char **argv) :
        EGS_AdvancedApplication(argc,argv), score(0), eflu(0), gflu(0), pheight(0), iregMaxHist(0),
        nreg(0), nph(0), Etot(0), rr_flag(0), current_weight(1), deflect_brems(false), RH_stream(0)`   


3 in 

`int Tutor7_Application::initScoring() {`

add 

` string ofile = constructIOFileName(".egsRHXYZE.xls",true);`
`RH_stream = ofstream(ofile.c_str(), ios::out);`    

Now you can use

`RH_stream << 

in the same way as 

cout <<
